[IMP] l10n_il: improve CoA, fiscal positions, taxes and translations.

hedshefer opened the pull request:
Description of the issue/feature this PR addresses:

* Fix wrong translations for taxes, tax grids, accounts and tax groups.
* Remove auto_apply for fiscal position israel, because partner from israel can also lead to other fiscal positions (self invoice, eilat,etc).
* Remove the duplicate account Income tax withheld - customers




--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

hedshefer commented:
@jco-odoo 
Can you please review this important PR?

hedshefer commented:
@jco-odoo 
@lordkrandel 
Hey, can you update me regarding the PR?

lordkrandel commented:
> @jco-odoo @lordkrandel Hey, can you update me regarding the PR?

I'm very sorry, both me and Josse are on vacation at the moment. (Josse will be back next week)
I've pinged our internal Accounting discord channel to let them know about your PR.

hedshefer commented:
@jco-odoo @mart-e @lordkrandel 
Can you update?

jco-odoo reviewed (commented):
- I don't see why the tags should suddenly be in Hebrew as they were in English and we have multi_lang through which we can put translations as well. 
- what is the purpose of the VAT Inputs Import Line Account?
- Better not to change the xml_id when not necessary for tax_group_vat_exempt

hedshefer commented:
> * I don't see why the tags should suddenly be in Hebrew as they were in English and we have multi_lang through which we can put translations as well.
Changed back to english
> * what is the purpose of the VAT Inputs Import Line Account?
Tax closing
> * Better not to change the xml_id when not necessary for tax_group_vat_exempt
Changed back to origin

hedshefer commented:
@jco-odoo @lordkrandel 
Can you please check it?

hedshefer commented:
@jco-odoo 
Hi,
I have fixed the PR to have the tags in english

jco-odoo commented on addons/l10n_il/data/account_tax_group_data.xml:
```diff
@@ -9,7 +9,7 @@
         <field name="country_id" ref="base.il"/>
     </record>
     <record id="tax_group_vat_exempt" model="account.tax.group">
-        <field name="name">VAT exempt</field>
```
No need to put sale

jco-odoo commented on addons/l10n_il/data/account_tax_template_data.xml:
```diff
@@ -79,7 +79,7 @@
     <record id="il_vat_sales_exempt" model="account.tax.template">
         <field name="sequence">9</field>
         <field name="description">0%</field>
-        <field name="name">VAT exempt</field>
```
No need to put sale as well as the tax is only used on the sale side already.

jco-odoo commented on addons/l10n_il/data/account_tax_template_data.xml:
```diff
@@ -355,7 +356,7 @@
     <record id="il_vat_purchase_exempt" model="account.tax.template">
         <field name="sequence">7</field>
         <field name="description">0%</field>
-        <field name="name">VAT exempt</field>
```
No need for adding purchases as the tax will already appear only on vendor bills.

jco-odoo commented on addons/l10n_il/data/fiscal_templates_data.xml:
```diff
@@ -5,7 +5,6 @@
         <field name="name">Israel</field>
         <field name="chart_template_id" ref="il_chart_template"/>
         <field name="country_id" ref="base.il"/>
-        <field name="auto_apply" eval="True"/>
```
Why?  If the partner needs another fiscal position, it can still be configured there and it can still be changed on the invoice.

hedshefer replied:
@jco-odoo 
What is the goal of an empty fiscal position which is auto apply like this one?

jco-odoo replied:
@hedshefer To be sure it does not apply the others.

hedshefer replied:
@jco-odoo 
fixed

jco-odoo commented on addons/l10n_il/data/account_account_tag.xml:
```diff
@@ -19,4 +19,8 @@
         <field name="name">Withholding Customers Tax Account</field>
     </record>
+
+    <record id="account_tag_vat_input_import_line" model="account.account.tag">
```
No way to use the VAT closing accounts on the tax group instead?  (or on the CoA)

hedshefer replied:
@jco-odoo 
reverted this tag change for now, until we have a better solution.

hedshefer commented:
@jco-odoo 
changed the import tax to division

jco-odoo commented on addons/l10n_il/data/account_tax_group_data.xml:
```diff
@@ -9,7 +9,7 @@
     </record>
     <record id="tax_group_vat_exempt" model="account.tax.group">
-        <field name="name">VAT exempt</field>
+        <field name="name">VAT exempt sale</field>
```
```suggestion
        <field name="name">VAT exempt</field>
```

jco-odoo commented on addons/l10n_il/data/account_tax_template_data.xml:
```diff
@@ -79,7 +79,7 @@
         <field name="sequence">9</field>
         <field name="description">0%</field>
-        <field name="name">VAT exempt</field>
+        <field name="name">VAT exempt sales</field>
```
```suggestion
        <field name="name">VAT exempt</field>
```

jco-odoo commented on addons/l10n_il/data/account_tax_template_data.xml:
```diff
@@ -355,7 +356,7 @@
         <field name="sequence">7</field>
         <field name="description">0%</field>
-        <field name="name">VAT exempt</field>
+        <field name="name">VAT exempt purchase</field>
```
```suggestion
        <field name="name">VAT exempt</field>
```

jco-odoo reviewed (commented):
I still do not understand why you want to add sale /purchase in those cases as it should be clear from the context.  (whether you are in a vendor bill or customer invoice) @hedshefer

hedshefer commented:
> I still do not understand why you want to add sale /purchase in those cases as it should be clear from the context. (whether you are in a vendor bill or customer invoice) @hedshefer

Just because it leads to wrong translations (in hebrew the purchase and sale exampts have different translation)

hedshefer commented:
@jco-odoo @lordkrandel 
Can we please merge this PR??

jco-odoo commented on addons/l10n_il/data/account_tax_template_data.xml:
```diff
@@ -392,33 +393,34 @@
+        <field name="amount_type">division</field>
         <field name="type_tax_use">purchase</field>
         <field name="chart_template_id" ref="il_chart_template"/>
         <field name="tax_group_id" ref="tax_group_vat_17"/>
```
@hedshefer The tax group here is weird but it was weird before.

hedshefer replied:
@jco-odoo Why is it weird?

jco-odoo replied:
because it is not a tax of 17% and you see it in the invoice tax summary as being 17% while the entire line is taken.  (I would be surprised that all of your import taxes would be 17%) Why not have a separate tax group for it?

hedshefer replied:
@jco-odoo 
This is how we report it in the tax report. The entire line included in the 17% input, no metter if it's taken from the import line or from the normal 17% input tax.

jco-odoo commented on addons/l10n_il/i18n/he_IL.po:
```diff
@@ -4,24 +4,34 @@
+msgstr ""
 
 #. module: l10n_il
+#: model:account.tax,description:l10n_il.1_il_vat_inputs_17
```
All these 1_il 2_il something are not needed.  Ideally, you just export the pot/po when the coa has not been installed yet in the db.

hedshefer replied:
@jco-odoo 
This is what I get when I export the PO from new DB.

jco-odoo replied:
@hedshefer It might be good to try it by installing another localization first, so the coa is not installed in the db, but the module is still there.

hedshefer replied:
@jco-odoo 
I don't understand. Some of the source terms are coming directly from l10n_il module, so in order to have the correct po it has to be installed, is't?

hedshefer replied:
@jco-odoo 
Done

jco-odoo commented on addons/l10n_il/i18n/l10n_il.pot:
```diff
@@ -809,12 +926,53 @@ msgid "Withholding Vendor Tax Account"
+
+#. module: l10n_il
+#: model:account.tax.report.line,tag_name:l10n_il.account_tax_report_line_out_base
+msgid "הכנסות חייבות במע\"מ"
```
Not sure where this comes from as the code in standard does not have the Hebrew originally.  (it should be in the po normally)

jco-odoo commented:
@robodoo r+
